src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -171,18 +170,18 @@ public Task ExecuteChildContentAsync()
         /// <returns>A <see cref="Task"/> that on completion returns the rendered child content.</returns>
         /// <remarks>
         /// Child content is only executed once. Successive calls to this method or successive executions of the 
-        /// returned <see cref="Task{string}"/> return a cached result.
+        /// returned <see cref="Task{TagHelperContent}"/> return a cached result.
         /// </remarks>
-        public async Task<string> GetChildContentAsync()
+        public async Task<TagHelperContent> GetChildContentAsync()
         {
             if (_childContent == null)
             {
-                _startWritingScope();
+                _startTagHelperWritingScope();
                 await _executeChildContentAsync();
-                _childContent = _endWritingScope().ToString();
+                _childContent = _endTagHelperWritingScope();
             }
 
-            return _childContent;
+            return new DefaultTagHelperContent().SetContent(_childContent);
```

Comment:
massive difference in scale @NTaylorMullen.  a `TagHelperContent` is quite small compared to the `string`s allocated when combining content on the page.

this is not the last change that will happen in Razor and if we see too many `TagHelperContent` allocations, we can lazy-load some and eliminate a few others.

side note: nothing about purpose number one is changing in this PR.